R/qra_fit_convenience.R

```diff
@@ -130,233 +66,132 @@ get_candidate_models <- function(
+  forecasts <- purrr::map_dfr(
+    monday_dates, 
+  covidHubUtils::load_latest_forecasts,
+    models = model_abbrs,
+    forecast_date_window_size = timezero_window_size,
+    locations = locations,
+    targets = targets,
+    types = c("point", "quantile"),
+    source = source,
+    hub_repo_path = hub_repo_path,
+    as_of = NULL,
+    hub = hub,
+    verbose = FALSE
+    )  %>% 
+  dplyr::mutate(
+    target_unadjusted = paste(horizon, temporal_resolution, "ahead", target_variable),
+    reference_date = covidEnsembles::calc_forecast_week_end_date(
+      forecast_date, 
+      target_unadjusted, 
+      return_type = "date"
+    ),
+    relative_horizon = covidEnsembles::calc_relative_horizon(
+      reference_date, 
+      target_end_date, 
+      target_unadjusted
+    )
+  ) %>% 
+  # keep only forecasts targeting dates after their reference dates
+  # and at relative horizons less than or equal to max_horizon
+  dplyr::filter(
+    relative_horizon <= max_horizon & relative_horizon > 0
+  )
 
   # Add in fake rows for models that submitted point forecasts but not quantile
   # forecasts -- this is done so those models will appear in the model
   # eligibility metadata
-  if (include_null_point_forecasts) {
-    point_forecasts <-
-      # get forecasts from local files
-      purrr::map_dfr(
-        tail(monday_dates, 1),
-        load_covid_forecasts,
-        model_abbrs = model_abbrs,
-        timezero_window_size = timezero_window_size,
-        types = "point",
-        targets = targets,
-        submissions_root = submissions_root,
-        keep_last = keep_last
-      ) %>%
-      # create columns for horizon, forecast week end date of the forecast, and
-      # target of the forecast
-      dplyr::mutate(
-        # the forecast_week_end_date variable is now mis-named; it represents the
-        # date relative to which the horizons and targets are defined for the
-        # purpose of building the ensemble.
-        forecast_week_end_date =
-          covidEnsembles::calc_forecast_week_end_date(timezero, target, return_type = "date"),
-        horizon =
-          covidEnsembles::calc_relative_horizon(forecast_week_end_date, target_end_date, target),
-        target =
-          covidEnsembles::calc_relative_target(forecast_week_end_date, target_end_date, target)
-      ) %>%
-      # keep only forecasts targeting dates after the forecast_week_end_date
-      # and less than the specified horizon relative to the forecast week end date
-      dplyr::filter(horizon > 0, horizon < UQ(horizon)) %>%
-      # keep only required locations and models for which no quantile forecasts
-      # were provided
-      dplyr::filter(
-        location %in% locations,
-        !(model %in% forecasts$model)
-      )
+  forecasts <- forecasts %>% dplyr::filter(
+    quantile %in% required_quantiles | type == "point"
+  ) %>% 
+  dplyr::group_by(model, forecast_date, location, target_end_date) %>% 
```

Comment:
```suggestion
  # within each combination of model, forecast_date, location, and target_end_date,
  # if both a quantile and point forecast were provided, drop the point forecast; else, keep both.
  # if the forecaster provided only a point forecast, convert it to a missing median
  # this is done so that the forecast will appear in validation outputs
  dplyr::group_by(model, forecast_date, location, target_end_date) %>% 
```